lib/style-transform.js

```diff
@@ -1,407 +1,45 @@
-                                        }
-
-                                        if (chr === ' ') {
-                                          if (piece === '') {
-                                            // ignore
-                                            continue
-                                          } else {
-                                            _line += piece + suffix + ' '
-                                            piece = ''
-                                          }
-                                        } else if (chr === ':') {
-                                          // pseudo-class or preudo-element
-                                          _line += piece + suffix
-                                          piece = chr
-                                          inCombinatorOrPseudo = true
-                                        } else if (chr === '+' || chr === '~' || chr === '>') {
-                                          // combinators
-                                          piece += chr
-                                          inCombinatorOrPseudo = true
-                                        } else {
-                                          piece += chr
-                                        }
-                                      }
-                                    }
-                                  }
-                                }
-
-                                // flush remainder
-                                if (piece.length) {
-                                  _line += piece
-                                  if (!inGlobal && !inCombinatorOrPseudo) {
-                                    _line += suffix
-                                  }
-                                }
-
-                                if (isLast) {
-                                  _line += '{'
-                                } else {
-                                  _line += ','
-                                }
-                            }
-
-                            line = _line;
-                        }
-                    }
-                }
-
-                generator && generator.addMapping({
-                    generated: {
-                      line: 1,
-                      column: output.length
-                    },
-                    source: fileName,
-                    original: {
-                        line: lineNumber,
-                        column: columnNumber
-                    }
-                })
-
-                output += line;
-                line = '';
-            }
-            // not `\t`, `\r`, `\n` characters
-            else if (code !== 9 && code !== 13 && code !== 10) {
-                line += styles[i];
-            }
-
-            // next character
-            i++;
-            columnNumber++;
+const stylis = require('stylis')
+
+function middleware(ctx, str, line, col, prefix) {
+if (ctx === 1.5) {
```

Comment:
Since the idea is to also add source-maps support will use a switch

```js
switch (ctx) {
    1.5: {
        // selector middleware
    }
    1:
    2: {
        // source-map generation
    }
}
```